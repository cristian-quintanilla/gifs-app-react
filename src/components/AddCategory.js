import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
	const [inputValue, setInputValue] = useState('');
	
	const handleInputChange = e => {
		setInputValue(e.target.value);
		// console.log('handleInputChange para pruebas');
	}

	const handleSubmit = e => {
		e.preventDefault();
		// console.log('handleSubmit para pruebas');
		
		// Validación
		if (inputValue.trim().length > 2){
			setCategories(categories => [ inputValue, ...categories ]);
			setInputValue('');
		}
	}

	return (
		<form onSubmit={ handleSubmit } >
			<p>{ inputValue }</p>
			<input
				type="text"
				placeholder="Agregar Categoría..."
				className="input--style-6"
				value={ inputValue }
				onChange={ handleInputChange } />
		</form>
	)
}

AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired
}