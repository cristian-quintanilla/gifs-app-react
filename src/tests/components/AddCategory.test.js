import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en el Componente <AddCategory />', () => {

	const setCategories = jest.fn();
	let wrapper = shallow( <AddCategory setCategories={ setCategories } /> );

	beforeEach(() => {
		jest.clearAllMocks();
		wrapper = shallow( <AddCategory setCategories={ setCategories } /> );
	});

	test('debe de mostrar el Componente correctamente.', () => {
		expect( wrapper ).toMatchSnapshot();
	});

	test('debe de cambiar el input', () => {
		const input = wrapper.find('input');
		const value = 'Hello World';

		input.simulate('change', {
			target: {
				value
			}
		});

		expect( wrapper.find('p').text().trim() ).toBe( value );

	});

	test('NO debe de postear la información con el submit', () => {
		wrapper.find('form').simulate('submit', {
			preventDefault(){}
		});

		expect( setCategories ).not.toHaveBeenCalled();
	});

	test('debe de llamar el setCategories y limpiar el input', () => {
		const input = wrapper.find('input');
		const value = 'Hello World';

		// 1. Simular el onChange en el input
		input.simulate('change', { target: { value } });

		// 2. Simular el submit
		wrapper.find('form').simulate('submit', {	preventDefault(){} });

		// 3. setCategories se debe de haber llamado
		expect( setCategories ).toHaveBeenCalled();
		// expect( setCategories ).toHaveBeenCalledTimes(2); -> Llamada dos veces
		// Esperar que se haya llamado con cualquier tipo de función
		expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );

		// 4. El valor del input debe de ser: ''
		expect( input.prop('value') ).toBe('');
	});

});