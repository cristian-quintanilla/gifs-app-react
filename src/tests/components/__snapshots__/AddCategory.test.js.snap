// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Componente <AddCategory /> debe de mostrar el Componente correctamente. 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    className="input--style-6"
    onChange={[Function]}
    placeholder="Agregar CategorÃ­a..."
    type="text"
    value=""
  />
</form>
`;
